src/python/gudhi/sklearn/cover_complex.py

```diff
@@ -0,0 +1,573 @@
+        f.write("}")
+        f.close()
+
+        L = np.linspace(epsv, 1.-epsv, 100)
+        colsrgb = []
+        try:
+            import colorsys
+            for c in L:
+                colsrgb.append(colorsys.hsv_to_rgb(c,1,1))
+            fig, ax = plt.subplots(figsize=(6, 1))
+            fig.subplots_adjust(bottom=0.5)
+            my_cmap = matplotlib.colors.ListedColormap(colsrgb, name=self.color_name)
+            cb = matplotlib.colorbar.ColorbarBase(ax, cmap=my_cmap, norm=matplotlib.colors.Normalize(vmin=minv, vmax=maxv), orientation="horizontal")
+            cb.set_label(self.color_name)
+            fig.savefig("colorbar_" + self.color_name + ".pdf", format="pdf")
+            plt.close()
+        except ImportError:
+            print("colorsys not found, colorbar not printed")
+
+    def print_to_txt(self):
+        """
+        Write the cover complex to a TXT file called "{self.input_name}.txt", that can be processed with KeplerMapper.
+        """
+        st = self.simplex_tree
+        f = open(self.input_name + ".txt", "w")
+        f.write(self.input_name + "\n")
+        f.write(self.cover_name + "\n")
+        f.write(self.color_name + "\n")
+        f.write("0 0\n")
+        f.write(str(st.num_vertices()) + " " + str(len(list(st.get_skeleton(1)))-st.num_vertices()) + "\n")
+        name2id = {}
+        idv = 0
+        for s,_ in st.get_skeleton(0):
+            f.write(str(idv) + " " + str(self.node_info[s[0]]["colors"][0]) + " " + str(self.node_info[s[0]]["size"]) + "\n")
+            name2id[s[0]] = idv
+            idv += 1
+        for s,_ in st.get_skeleton(1):
+            if len(s) == 2:
+                f.write(str(name2id[s[0]]) + " " + str(name2id[s[1]]) + "\n")
+        f.close()
+
+    class _constant_clustering():
+        def fit_predict(X):
+            return np.zeros([len(X)], dtype=np.int32)
+
+
+class MapperComplex(CoverComplexPy, BaseEstimator, TransformerMixin):
+    """
+    This is a class for computing Mapper simplicial complexes on point clouds or distance matrices.
+    """
+    def __init__(self, input_type="point cloud", colors=None, mask=0,
+                       filters=None, filter_bnds=None, resolutions=None, gains=None, clustering=DBSCAN(), N=100, beta=0., C=10.,
+                       input_name="data", cover_name="cover", color_name="color", verbose=False):
+        """
+        Constructor for the MapperComplex class.
+
+        Parameters:
+            input_type (string): type of input data. Either "point cloud" or "distance matrix".
+            colors (numpy array of shape (num_points) x (num_colors)): functions used to color the nodes of the cover complex. More specifically, coloring is done by computing the means of these functions on the subpopulations corresponding to each node. If None, first coordinate is used if input is point cloud, and eccentricity is used if input is distance matrix.
```

Comment:
`colors` and `filters` are both specific to the point set (they know num_points) that will be passed to `fit`. It seems to me that having to pass them to the constructor goes completely against the scikit-learn philosophy and they should instead be passed to `fit` (https://scikit-learn.org/stable/developers/develop.html#fitting says it is normal to have extra data-dependent arguments in `fit`).
(Or we could give up the pretense that these classes are sklearn-compatible.)

Reply:
I agree. When I wrote that code I was still thinking of the filter and color as parameters, but it makes more sense to call them at the `fit` step. This also avoids having to initialize the class with something that has to be the same size of the data before even providing the data itself. I also moved the `assignments` param of `NerveComplex` to `fit` for the same reason.

Reply:
You forgot to update the example in the doc cover_complex_sklearn_isk_ref.rst, so the tests fail.

Reply:
Fixed!